File: term4/Kurs-projektowania-aplikacji-z-bazami-danych/project-old/src/main/kotlin/dbapp/project/infrastructure/test/TestOrderRepository.kt
package dbapp.project.infrastructure.test

import dbapp.project.common.GenericTestRepository
import dbapp.project.domain.order.Order
import dbapp.project.domain.order.OrderItem
import dbapp.project.domain.order.OrderRepository

class TestOrderRepository : GenericTestRepository<Order>(
        mutableSetOf(
            Order(1, 1, 1, setOf(
                    OrderItem(1, 2137, 2))),
            Order(2, 2, 2, setOf(
                    OrderItem(1, 2137, 2),
                    OrderItem(2, 2137, 5)))
    )
), OrderRepository {

    override fun findAllOfCustomer(customerID: Int) = objects.filter { it.customerID == customerID }
}

File: term4/Kurs-projektowania-aplikacji-z-bazami-danych/project/Store/Store.Infrastructure/Repositories/OrderRepository/TestOrderRepository.cs
using Store.Common;
using Store.Domain.Model.Order;
using System.Collections.Generic;
using System.Linq;

namespace Store.Infrastructure.Repositories.OrderRepository
{
    public class TestOrderRepository : GenericTestRepository<Order>, IOrderRepository
    {
        private ICollection<Order> Orders { get; }

        public TestOrderRepository() : this(new List<Order> {
            new Order { Id = 1, CustomerId = 1, ShipmentId = 1, OrderItems = new HashSet<OrderItem>() },
            new Order { Id = 2, CustomerId = 2, ShipmentId = 2, OrderItems = new HashSet<OrderItem>() },
            new Order { Id = 3, CustomerId = 3, ShipmentId = 3, OrderItems = new HashSet<OrderItem>() }
        }) { }

        private TestOrderRepository(ICollection<Order> orders) : base(orders)
        {
            Orders = orders;
        }

        public IEnumerable<Order> FindAllOfCustomer(int customerId)
        {
            return Orders.Where(o => o.CustomerId == customerId);
        }
    }
}
